arbitrum-docs/devs-how-tos/how-to-estimate-gas.mdx

```diff
@@ -0,0 +1,135 @@
+---
+id: how-to-estimate-gas
+title: "How to estimate gas in Arbitrum"
+description: Learn how to estimate gas before submitting transactions.
+author: TucksonDev
+---
+
+import PublicPreviewBannerPartial from '../partials/_public-preview-banner-partial.md'; 
+
+<PublicPreviewBannerPartial />
+
+This how-to is intended for users and developers who want to understand how gas is calculated in Arbitrum, and how it can be estimated before submitting transactions. More information about the theory behind these calculations can be found in this [Medium article](https://medium.com/offchainlabs/understanding-arbitrum-2-dimensional-fees-fd1d582596c9) and the [Gas and Fees](https://developer.arbitrum.io/arbos/gas) page. 
+
+We will first break down the formula mentioned in the Medium article, moving then to where to get the information of each variable, and finally seeing an example of how to apply the formula in your code as well as other practical ways of estimating gas costs.
+
+However, if you want to jump straight to the code, we have created [this script in our tutorials repository](https://github.com/OffchainLabs/arbitrum-tutorials/tree/master/packages/gas-estimation) that goes through all the calculations explained in this how-to.
+
+## Breaking down the formula
+
+As explained in the Medium article, the transaction fees to pay at any given moment are the result of the following product:
+
+```
+Transaction fees (TXFEES) = L2 Gas Price (P) * Gas Limit (G)
+```
+
+This Gas Limit includes the gas of the L2 computation and an additional buffer to cover the gas to be paid when posting the batch including this transaction on L1.
+
+```
+Gas Limit (G) = Gas used on L2 (L2G) + Extra Buffer for L1 cost (B)
+```
+
+This buffer takes into account the cost of posting the transaction, batched and compressed, on L1. This L1 posting cost is calculated as follows:
+
+```
+L1 Cost (L1C) = L1 price per byte of data (L1P) * Size of data to be posted in bytes (L1S)
+```
+
+To calculate the buffer, that cost is divided by the L2 Gas Price.
+
+```
+Extra Buffer (B) = L1 Cost (L1C) / L2 Gas Price (P)
+```
+
+Finally, using all these elements, the formula can be written as follows:
+
+```
+TXFEES = P * (L2G + ((L1P * L1S) / P))
+```
+
+## Where do we get all this information from?
+
+We will use two resources available in Arbitrum: the [ArbGasInfo](https://arbiscan.io/address/0x000000000000000000000000000000000000006c) precompile and the [NodeInterface](https://developer.arbitrum.io/arbos/gas#nodeinterfacesol).
+
+- P (L2 Gas Price) ⇒ Price to pay for each gas unit. It starts at 0.1 gwei and can increase depending on the demand for network resources.
+    - Call `ArbGasInfo.getPricesInWei()` and get the sixth element
```

Comment:
these 3 all return the same value, right? As below, I think it's best to just include one method